math/types.ts

```diff
@@ -0,0 +1,264 @@
+import { Point } from "../handles/point";
+
+// Converts Degrees to Radians
+const DEGTORAD = 0.017453293;
+// Converts Radians to Degrees
+const RADTODEG = 57.295779513;
+
+/**
+ *  Encapsulates an angle, abstracting the usage of degrees and radians.
+ */
+export class Angle {
+  protected constructor(private readonly rads: number) {}
+
+  public static fromDegrees(degrees: number) {
+    return new Angle(degrees * DEGTORAD);
+  }
+
+  public static fromRadians(radians: number) {
+    return new Angle(radians);
+  }
+
+  public static random() {
+    return new Angle(GetRandomReal(0, math.pi * 2));
+  }
+
+  public get degrees() {
+    return this.rads * RADTODEG;
+  }
+
+  public get radians() {
+    return this.rads;
+  }
+
+  public get cos() {
+    return Cos(this.rads);
+  }
+
+  public get sin() {
+    return Sin(this.rads);
+  }
+
+  public add(other: Angle) {
+    return new Angle(this.radians + other.radians);
+  }
+
+  // returns a unit length Vec2 in the direction of this angle,
+  // parallel to the ground plane.
+  public asDirection() {
+    return new Vec2(this.cos, this.sin);
+  }
+}
+
+export const degrees = Angle.fromDegrees;
+export const radians = Angle.fromRadians;
+export const randomAngle = Angle.random;
+
+export class Vec2 {
```

Comment:
Why are all classes in one file? Should be split.

Reply:
I wrote these classes originally in one file, but for some reason when I added the `angleTo` method on the vec2, games were crashing before getting to the load screen (I think this means invalid Lua in the map's Lua). I spent some time trying to figure it out, but nothing solved the issue except moving them into the same file. 

If you have some ideas on why this was happening, I'm all ears to try again.